### ISSUE TITLE
Mocking methods with optional parameters does not seem to work

### ISSUE THREAD

**Issue by sgronblo** [NagRock/ts-mockito#36]
```
import { instance, mock, when } from 'ts-mockito'
import expect from 'ceylon'

export default class TestClass {
    withOptionalParameter(foo: string, bar?: string) {
        return foo + bar
    }
}

describe('using a ts-mockito mock of TestClass ->', () => {
    const mockCreator = mock(TestClass)
    when(mockCreator.withOptionalParameter('a')).thenReturn('aa')
    when(mockCreator.withOptionalParameter('b', 'c')).thenReturn('bcbc')
    const mockInstance = instance(mockCreator)
    describe('withOptionalParameter', () => {
        it('should work!', () => {
            expect(mockInstance.withOptionalParameter('a')).toEqual('aa')
            expect(mockInstance.withOptionalParameter('b', 'c')).toEqual('bcbc')
        })
    })
})
```

This results in a "TypeError: Cannot read property 'match' of undefined" when running the test.

If the first `when` call is change to have two parameters like the second one, ie `when(mockCreator.withOptionalParameter('a', 'a')).thenReturn('aa')` then the test works. So apparently something goes wrong if you provide multiple `when` matchers with different numbers of parameters.

**Comment by sgronblo** [comment]
Figured out a workaround of specifying an explicit `undefined` for the optional parameter in the `when` call

**Comment by NagRock** [comment]
Thanks for reporting @sgronblo. I will take a look at it.

**Comment by NagRock** [comment]
@sgronblo issue that you have reported should be fixed in 2.1.1 version:
https://github.com/NagRock/ts-mockito/releases/tag/v2.1.1

Once again thanks for reporting this!
